recipes/mold/all/test_package/conanfile.py

```diff
@@ -1,10 +1,16 @@
-import os
-from conans import ConanFile, tools
-from conan.tools.build import cross_building
+from conan import ConanFile
+from conan.tools.build import can_run
 
-class TestPackageConan(ConanFile):
-    settings = "os", "arch", "build_type", "compiler"
+
+class MoldTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "VirtualBuildEnv"
+    test_type = "explicit"
+
+    def build_requirements(self):
+        self.tool_requires(self.tested_reference_str)
 
     def test(self):
-        if not cross_building(self):
-            self.run("mold -v", run_environment=True)
+        if can_run(self):
+            self.run("mold -v")
```

Comment:
@uilianries, please try your changes locally first. They break this recipe.

Reply:
So something is really wrong, because it should work normally for any tool.

Reply:
@uilianries, why ninja has it the same way then?
https://github.com/conan-io/conan-center-index/blob/master/recipes/ninja/all/test_package/conanfile.py

Reply:
Can we have mold recipe "as is" and rise a separate discussion about using VirtualBuildEnv / VirtualRunEnv for build tools?

Reply:
@AndreyMlashkin Not needed. Indeed `VirtualBuildEnv` is correct one.

As we need to run a tool from the package folder, we need that build env, only in case running a binary from test package we should use `VirtualRunEnv`.

For instance, the `VirtualRunEnv` will produce `conanrunenv-release-x86_64.sh `, which contains:

```
echo "echo Restoring environment" >> "/home/conan/project/test_package/build/8130ca333cc035689d4db02db920b56a5f995fbb/deactivate_conanrunenv-release-x86_64.sh"
for v in 
do
    is_defined="true"
    value=$(printenv $v) || is_defined="" || true
    if [ -n "$value" ] || [ -n "$is_defined" ]
    then
        echo export "$v='$value'" >> "/home/conan/project/test_package/build/8130ca333cc035689d4db02db920b56a5f995fbb/deactivate_conanrunenv-release-x86_64.sh"
    else
        echo unset $v >> "/home/conan/project/test_package/build/8130ca333cc035689d4db02db920b56a5f995fbb/deactivate_conanrunenv-release-x86_64.sh"
    fi
done
```

As you can see, it does not set `PATH` neither `LD_PATH` on Linux.

On the other hand, `VirtualBuildEnv` provides everything:

```
echo "echo Restoring environment" >> "/home/conan/project/test_package/build/8130ca333cc035689d4db02db920b56a5f995fbb/deactivate_conanbuildenv-release-x86_64.sh"
for v in MOLD_ROOT PATH LD_LIBRARY_PATH DYLD_LIBRARY_PATH
do
    is_defined="true"
    value=$(printenv $v) || is_defined="" || true
    if [ -n "$value" ] || [ -n "$is_defined" ]
    then
        echo export "$v='$value'" >> "/home/conan/project/test_package/build/8130ca333cc035689d4db02db920b56a5f995fbb/deactivate_conanbuildenv-release-x86_64.sh"
    else
        echo unset $v >> "/home/conan/project/test_package/build/8130ca333cc035689d4db02db920b56a5f995fbb/deactivate_conanbuildenv-release-x86_64.sh"
    fi
done


export MOLD_ROOT="/home/conan/.conan/data/mold/1.4.2/_/_/package/61177d595b541b64453bdb404af2a28944369c40/bin:$MOLD_ROOT"
export PATH="/home/conan/.conan/data/mold/1.4.2/_/_/package/61177d595b541b64453bdb404af2a28944369c40/bin:/home/conan/.conan/data/xxhash/0.8.1/_/_/package/d750cfef2639f696140dcb4782e5fd0eb44bb369/bin:/home/conan/.conan/data/openssl/1.1.1q/_/_/package/19729b9559f3ae196cad45cb2b97468ccb75dcd1/bin:$PATH"
export LD_LIBRARY_PATH="/home/conan/.conan/data/onetbb/2021.3.0/_/_/package/79f200efe18f84353bb1264b3eeee74874c3dd69/lib:/home/conan/.conan/data/xxhash/0.8.1/_/_/package/d750cfef2639f696140dcb4782e5fd0eb44bb369/lib:/home/conan/.conan/data/openssl/1.1.1q/_/_/package/19729b9559f3ae196cad45cb2b97468ccb75dcd1/lib:/home/conan/.conan/data/zlib/1.2.12/_/_/package/19729b9559f3ae196cad45cb2b97468ccb75dcd1/lib:$LD_LIBRARY_PATH"
export DYLD_LIBRARY_PATH="/home/conan/.conan/data/onetbb/2021.3.0/_/_/package/79f200efe18f84353bb1264b3eeee74874c3dd69/lib:/home/conan/.conan/data/xxhash/0.8.1/_/_/package/d750cfef2639f696140dcb4782e5fd0eb44bb369/lib:/home/conan/.conan/data/openssl/1.1.1q/_/_/package/19729b9559f3ae196cad45cb2b97468ccb75dcd1/lib:/home/conan/.conan/data/zlib/1.2.12/_/_/package/19729b9559f3ae196cad45cb2b97468ccb75dcd1/lib:$DYLD_LIBRARY_PATH"
```